IssueTracker/IssueTracker.FileSystem/Repository/BlobData.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Runtime.CompilerServices;
+using Azure.Storage.Blobs;
+using Azure.Storage.Sas;
+
+[assembly: InternalsVisibleTo("IssueTracker.UnitTest")]
+namespace IssueTracker.FileSystem;
+public class BlobData : IBlobProvider
+{
+    private readonly IBlobConfigurationFactory _config;
+    private BlobServiceClient BlobServiceClient { get; set; }
+    private BlobContainerClient ContainerClient { get; set; }
+    internal BlobData(IBlobConfigurationFactory config)
+    {
+        _config = config;
+        BlobServiceClient = new BlobServiceClient(_config.ConnectionString);
+        ContainerClient = BlobServiceClient.GetBlobContainerClient(_config.Container);
+    }
+
+    public Task<IEnumerable<Models.File>> GetFilesAsync(IEnumerable<Models.File> files)
+    {
+        var fileResult = new List<Models.File>();
+        if (files.Count() <= 0)
+            throw new ArgumentException("I can't get nothing, files is empty!");
+        foreach (var file in files)
+        {
+            var fileAtachment = new Models.File();
+            if (file.Id == null || file.Id == string.Empty)
+                throw new ArgumentException("Invalid file id!");
+            if (file.Extension == null || file.Extension == string.Empty)
+                throw new ArgumentException("Invalid file extension!");
+            fileAtachment.Id = file.Id;
+            fileAtachment.Extension = file.Extension;
+            BlobClient information = ContainerClient.GetBlobClient(file.Id + file.Extension);
+            fileAtachment.Link = GetBlobSasUri(information).ToString();
+            if (fileAtachment.Link == null)
+                throw new ArgumentException($"Sas Invalid for {file.Name}!");
```

Comment:
This should not be an ArgumentException.

Reply:
will change in the next feature: ErrorManagement